src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ModuleTokenResolver.cs

```diff
@@ -246,6 +247,13 @@ private void AddModuleTokenForFieldReference(TypeDesc owningType, ModuleToken to
         // be consistent in all runs of the compiler
         void SetModuleTokenForTypeSystemEntity<T>(ConcurrentDictionary<T, ModuleToken> dictionary, T tse, ModuleToken token)
         {
+            // We can only use tokens from the manifest mutable module or from one of the modules that versions
+            // with the current compilation. Any other module tokens may change while the code executes
+            if (token.Module != _manifestMutableModule && _compilationModuleGroup.VersionsWithModule((ModuleDesc)token.Module));
```

Comment:
```suggestion
            if (token.Module != _manifestMutableModule && _compilationModuleGroup.VersionsWithModule((ModuleDesc)token.Module))
```

Reply:
Sigh. This is what I get for refactoring just before committing.